eth/handler.go

```diff
@@ -288,22 +292,68 @@ func newHandler(config *handlerConfig) (*handler, error) {
 	return h, nil
 }
 
+// protoTracker tracks the number of active protocol handlers.
+func (h *handler) protoTracker() {
+	defer h.wg.Done()
+	var (
+		active int
+		done   = make(chan struct{})
+	)
+	// Start goroutine to track active protocol handlers.
+	go func() {
+		for {
+			select {
+			case <-h.handlerStartCh:
+				active--
```

Comment:
This probably works fine, but it's a bit funny, you are counting on the negative side of the x-axis :)

Reply:
Heh oops - fixed now

Reply:
Oh, another thing though, this is "probably fine" but strictly speaking, this is racy. You are accessing `active` from two different go-routines simultaneously, which is a no-no. So either use an atomic int, or use an int, but make it scoped inside this goroutine, and let the outer checker wait on a Cond instead.